src/bloqade/squin/op/stmts.py

```diff
@@ -87,15 +109,18 @@ class Control(CompositeOp):
 
 @statement(dialect=dialect)
 class Rot(CompositeOp):
```

Comment:
Is there a reason `Rot` doesn't have any hermitian trait?

Reply:
No not really. I can add it, but it's only hermitian if the angle is 0, in which case it's actually just the identity. So the use-case is just very limited, which is why I skipped it.

Reply:
Oh, this also applies to `U3`, `PhaseOp` and `ShiftOp` (maybe a few more). I feel like it's a bit of an overkill to have the hermitian analysis in place here, as it's always only the case that they are hermitian when the exponent is 0 and therefore the operator is just the identity. What do you think @johnzl-777 ?

As a side note: I generally think this should be a rewrite pass, which just replaces the operator by the identity directly. But we don't really have any such simplification passes in-place right now. For example, simplifying multiplications with identity operators would be another such candidate.

Reply:
I see, I think your proposal for a rewrite makes sense here instead of burdening other statements with the hermitian information. I suppose I'm just nervous how much of a downside we could get if it turns out something IS hermitian (this 0 exponent case) and we don't catch it. 

I imagine it wouldn't be the worst thing in the world and if we knew we'd have something spitting out 0-exponents we'd most likely have thought to include that rewrite in the first place.